### PR TITLE
Updated timeline documentation and examples to be carbon centric

### LINKED ISSUE #173: Update documentation for Timeline component to be carbon-centric
# Story

## Description

As a current consumer of carbon, I'd like for the public facing documentation for the Timeline component to utilize carbon until terra-graphs is released

## Motivation

So that I will have the information I need to use the Timeline component from carbon as things currently are released

## Definition of Done

We'll know we're done when the documentation in terra-graphs-docs is utilizing carbon instead of terra-graphs.

### PR DESCRIPTION
**Pull request by AshishMotanamGurunadham**
### Summary

- Made changes to the documentation for all Timeline graph examples to make the dev site carbon-centric.

Closes #173 

Thank you for contributing to Terra.
@cerner/terra
@cerner/carbon

### CODE REVIEW

File: packages/terra-graphs-docs/CHANGELOG.md [hunk 1]
```diff
@@ -30,3 +30,4 @@
   * Updated general graph documentation and examples to be carbon-centric.
   * Updated Paired Result documentation and examples to be carbon-centric.
   * Updated Pie chart documentation and examples to be carbon-centric..
+  * Updated Timeline documentation and examples to be carbon-centric..
```
Comment (sdadn) [inline packages/terra-graphs-docs/CHANGELOG.md:33]: nit: double periods at the end

File: packages/terra-graphs-docs/src/terra-dev-site/doc/example/Graphs-Carbon/Timeline/BasicTimeline/SimpleTimeline.jsx [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+import Carbon from '@cerner/carbon-graphs/lib/js/carbon';
+import '@cerner/terra-graphs-docs/lib/Css/ExampleGraphContainer.module.scss';
+import '@cerner/terra-graphs/lib/components/Graph.module.scss';
+import '@cerner/terra-graphs/lib/components/Timeline/Timeline.module.scss';
+import getTimelineConfig from '@cerner/terra-graphs-docs/lib/example-datasets/graphConfigObjects/Timeline/timelineConfigObject';
+import exampleData from '@cerner/terra-graphs-docs/lib/example-datasets/dataObjects/Timeline/defaultDataset.js';
+
+/*
+Please refer to the documentation below to see the graphConfig and data objects
+*/
```
Comment (sdadn) [inline packages/terra-graphs-docs/src/terra-dev-site/doc/example/Graphs-Carbon/Timeline/BasicTimeline/SimpleTimeline.jsx:12]: nit: add blank line for consistency with other examples

File: packages/terra-graphs-docs/src/terra-dev-site/doc/example/Graphs-Carbon/Timeline/BasicTimeline/AxisWithNoTickValues.jsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+import Carbon from '@cerner/carbon-graphs/lib/js/carbon';
+import '@cerner/terra-graphs-docs/lib/Css/ExampleGraphContainer.module.scss';
+import '@cerner/terra-graphs/lib/components/Graph.module.scss';
+import '@cerner/terra-graphs/lib/components/Timeline/Timeline.module.scss';
+import getTimelineConfig from '@cerner/terra-graphs-docs/lib/example-datasets/graphConfigObjects/Timeline/timelineConfigObject';
+import exampleData from '@cerner/terra-graphs-docs/lib/example-datasets/dataObjects/Timeline/defaultDataset.js';
+
+/*
+Please refer to the documentation below to see the graphConfig and data objects
+*/
```
Comment (sdadn) [inline packages/terra-graphs-docs/src/terra-dev-site/doc/example/Graphs-Carbon/Timeline/BasicTimeline/AxisWithNoTickValues.jsx:12]: nit: new line for consistency

File: packages/terra-graphs-docs/src/terra-dev-site/doc/Graphs/Timeline/BasicTimeline/AxisWithNoTickValues.2.doc.mdx [hunk 4]
```diff
@@ -11,21 +10,17 @@ This is a basic timeline graph with no axis ticks.
 ## Getting Started
 
 - Install with [npmjs](https://www.npmjs.com):
-  - `npm install terra-graphs`
+  - `npm i @cerner/carbon-graphs --save-dev`
```
Comment (tbiethman) [inline packages/terra-graphs-docs/src/terra-dev-site/doc/Graphs/Timeline/BasicTimeline/AxisWithNoTickValues.2.doc.mdx:13]: Should we be suggesting consumers install this package as a devDependency?

Comment (AshishMotanamGurunadham) [reply packages/terra-graphs-docs/src/terra-dev-site/doc/Graphs/Timeline/BasicTimeline/AxisWithNoTickValues.2.doc.mdx:13]: yes Tyler, that is how we are suggesting consumers even before we merged into terra.

File: packages/terra-graphs-docs/src/terra-dev-site/doc/example/Graphs-Carbon/Timeline/Reflow/DynamicData.jsx [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+        hour = panState.initial - panState.factor;
+        break;
+      case 'panRight':
+        hour = panState.initial + panState.factor;
+        break;
+      default:
+        return panState;
+    }
+
+    graph.config.axis.x.lowerLimit = new Date(2016, 0, 1, hour).toISOString();
```
Comment (tbiethman) [inline packages/terra-graphs-docs/src/terra-dev-site/doc/example/Graphs-Carbon/Timeline/Reflow/DynamicData.jsx:48]: Typically, your reducer shouldn't have any side effects. The reducer should determine and return the new state; you could add a separate effect for other changes.

```jsx
const [state, dispatch] = React.useReducer(reducer, initialState);

React.useLayoutEffect(() => {
  if (!graph) { return; }
  
  graph.config.axis.x.lowerLimit = new Date(2016, 0, 1, state.initial).toISOString();
  graph.config.axis.x.upperLimit = new Date(2016, 0, 2, state.initial).toISOString();
  
  const newDataset = {
    panData: utils.deepClone(data2),
  };

  graph.reflowMultipleDatasets(newDataset);
}, [graph, state.initial]);
```

Comment (AshishMotanamGurunadham) [reply packages/terra-graphs-docs/src/terra-dev-site/doc/example/Graphs-Carbon/Timeline/Reflow/DynamicData.jsx]: Fixed [here](https://github.com/cerner/terra-graphs/pull/193/commits/4f472d55aa757d4dd6e033b6e1c90aa0dcc4095a).

**Review by tbiethman — APPROVED** [review]

**Review by sdadn — APPROVED** [review]

**Pull request merged**